Automattic/woocommerce-payments
# Updates to use centralised account status function

Fixes #7655

#### Changes proposed in this Pull Request



* Introduces new function relying on `details_submitted` flag (and not checking Stripe, making the assumption this should be handled higher in the code).
* Update to use this function instead of `partially_onboarded` / `complete` functions.
* Some refactors.







#### Testing instructions





In order to test this we mainly need to try out a few scenarios with non-onboarded accounts and make sure they work as expected. For example:

* Run `npm run listen` on the server.
* Start a new onboarding. When you hit the Stripe KYC, go back to WooPayments (so the onboarding is only partially completed).
* You should only be able to see the Overview page of WooPayments, with other pages not available in the dropdown.
* Try going to the settings page: http://host.docker.internal:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments
* You should be redirected to the Overview page.
* Complete the onboarding. Verify that you can now access the Settings page via `Payments > Settings` (and also check that other pages such as Transactions load successfully.

-------------------


**Post merge**
